pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -1509,41 +1508,40 @@ func TestMergePullSecrets(t *testing.T) {
 		},
 		{
 			name: "Test should fail as local an global pull secret is not available",
-			existingCD: []runtime.Object{
-				func() *hivev1.ClusterDeployment {
-					return getCDWithoutPullSecret()
-				}(),
+			existingObjs: []runtime.Object{
+				getCDWithoutPullSecret(),
 			},
 			expectedErr: true,
 		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			fakeClient := fake.NewFakeClient(test.existingCD...)
+			if test.globalPullSecret != "" {
+				globalPullSecretObj := createGlobalPullSecretObj(corev1.SecretTypeDockerConfigJson, globalPullSecret, corev1.DockerConfigJsonKey, test.globalPullSecret)
+				test.existingObjs = append(test.existingObjs, globalPullSecretObj)
+			}
+			if test.localPullSecret != "" {
+				localSecretObject := testSecret(corev1.SecretTypeDockercfg, pullSecretSecret, corev1.DockerConfigJsonKey, test.localPullSecret)
+				test.existingObjs = append(test.existingObjs, localSecretObject)
+			}
+			fakeClient := fake.NewFakeClient(test.existingObjs...)
 			rcd := &ReconcileClusterDeployment{
 				Client:                        fakeClient,
 				scheme:                        scheme.Scheme,
 				logger:                        log.WithField("controller", "clusterDeployment"),
 				remoteClusterAPIClientBuilder: testRemoteClusterAPIClientBuilder,
 			}
 
-			localSecretObject := testSecret(corev1.SecretTypeDockercfg, pullSecretSecret, corev1.DockerConfigJsonKey, test.localPullSecret)
-			err := rcd.Create(context.TODO(), localSecretObject)
-			if err != nil {
-				t.Errorf("unexpected error: %v", err)
-				return
-			}
 			cd := getCDFromClient(rcd.Client)
 			if test.globalPullSecret != "" {
-				os.Setenv("GLOBAL_PULL_SECRET", test.globalPullSecret)
+				os.Setenv(constants.GlobalPullSecret, globalPullSecret)
 			}
-			defer os.Unsetenv("GLOBAL_PULL_SECRET")
+			defer os.Unsetenv(constants.GlobalPullSecret)
+
 			expetedPullSecret, err := rcd.mergePullSecrets(cd, rcd.logger)
 			if test.expectedErr {
```

Comment:
If the test does not expect an error, then the test should assert that there was not an error.